### ISSUE TITLE
Why cmd.run for docker_image.present ?

### ISSUE THREAD

**Issue by Poil** [saltstack-formulas/docker-formula#191]
Hi,

Why we do this
```
{% for name, container in containers.items() %}
docker-image-{{ name }}:
  cmd.run:
    - name: docker pull {{ container.image }}
    - require:
      - service: docker-service
```

And not
```
{% for name, container in containers.items() %}
docker-image-{{ name }}:
  docker_image.present:
    - name: {{ container.image }}
    - tag: {{ container.tag }}
    - force {{ container.force_pull }} # allow to choose when doing a docker pull or not if image already exists locally
    - require:
      - service: docker-service
```

Best regards

**Comment by noelmcloughlin** [comment]
Hi @Poil 

This is bug I guess  - can you update your PR to fix this?

Probably docker_image.present did not exist when original PR was opened few years ago.
https://github.com/saltstack-formulas/docker-formula/pull/46/files#diff-f5f82bd5b538aa8dc056f6df012cc707

**Comment by Poil** [comment]
Hi,
I've start working on it but now what I don't understand is what is `compose-ng.sls`, it's not using docker-compose but use salt docker state to manage the container, so it look like what I want to do on `containers.sls`. (also `compose-ng.sls` include `compose.sls` that install docker-compose, but I don't understand why)

I don't see any dependency on compose on the salt docker state, but perhaps I'm wrong ?

**Comment by noelmcloughlin** [comment]
There is some info the README.  This formula is muddled in some places so do your best;-) 
https://github.com/saltstack-formulas/docker-formula#dockercompose-ng
I think this is two formulas in one:  docker and docker.ng

**Comment by noelmcloughlin** [comment]
Fixed in merged #256
